### PR TITLE
Add possibility to add custom factories tagged services.

### PR DESCRIPTION
**Pull request by chosroes**
Should do the job for #115

### CODE REVIEW

**Comment by bytehead** [comment]
I'm not sure, if this changes anything?  
Some code just moved from `getFactories()` method into `load()` method...

**Comment by bytehead** [comment]
Nevermind. I think i got it 😄 .

**Comment by chosroes** [comment]
@bytehead I changed my commit earlier, could this have made the travis build fail ?

**Comment by bytehead** [comment]
Nope, you have to fix it here: https://github.com/chosroes/OneupFlysystemBundle/blob/master/DependencyInjection/OneupFlysystemExtension.php#L52 (pass the `$container` into `$this->getFactories()`).

**Comment by chosroes** [comment]
Yes, I missed it...

**Comment by bytehead** [comment]
Will you fix it? Otherwise you should click `Allow edits from maintainers.` :)

**Comment by chosroes** [comment]
@bytehead I fixed it

**Comment by bytehead** [comment]
Can't see the change. Did you push?

File: DependencyInjection/OneupFlysystemExtension.php [hunk 1]
```diff
@@ -49,7 +51,7 @@ public function load(array $configs, ContainerBuilder $container)
 
     public function getConfiguration(array $config, ContainerBuilder $container)
     {
-        list($adapterFactories, $cacheFactories) = $this->getFactories();
+        list($adapterFactories, $cacheFactories) = $this->getFactories($container);
```
Comment (chosroes) [inline DependencyInjection/OneupFlysystemExtension.php]: Here

**Comment by bytehead** [comment]
Thank you!

**Pull request merged**